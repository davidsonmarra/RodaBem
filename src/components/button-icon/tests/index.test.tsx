import React from 'react';
import {create} from 'react-test-renderer';
import {AddIcon} from '@assets';
import ButtonIcon from '..';

const createComponent = ({onPress = jest.fn(), children = <AddIcon />}) =>
  create(<ButtonIcon onPress={onPress}>{children}</ButtonIcon>);

describe('Button', () => {
  it('renders correctly', () => {
    const tree = createComponent({}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders children correctly', () => {
    const instance = createComponent({}).root;
    expect(instance.props.children).toStrictEqual(<AddIcon />);
  });

  it('should call onPress', () => {
    const mockOnPress = jest.fn();
    const instance = createComponent({onPress: mockOnPress}).root;
    instance.props.onPress();
    expect(mockOnPress).toHaveBeenCalled();
  });
});
