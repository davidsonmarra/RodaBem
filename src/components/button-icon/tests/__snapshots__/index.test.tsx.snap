// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#E30B0B",
      "borderRadius": 43,
      "height": 64,
      "justifyContent": "center",
      "opacity": 1,
      "width": 64,
    }
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="48"
    bbWidth="48"
    fill="#FFFFFF"
    focusable={false}
    height="48"
    meetOrSlice={0}
    minX={0}
    minY={-960}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 48,
          "width": 48,
        },
      ]
    }
    vbHeight={960}
    vbWidth={960}
    width="48"
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4294967295,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M440-440H200v-80h240v-240h80v240h240v80H520v240h-80v-240z"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
